floors/validate.go

```diff
@@ -47,10 +75,26 @@ func validateFloorParams(extFloorRules *openrtb_ext.PriceFloorRules) error {
 	return nil
 }
 
-func selectValidFloorModelGroups(modelGroups []openrtb_ext.PriceFloorModelGroup) ([]openrtb_ext.PriceFloorModelGroup, []error) {
+// selectValidFloorModelGroups validates each modelgroup for SkipRate and ModelGroup and drops invalid modelGroups
+func selectValidFloorModelGroups(modelGroups []openrtb_ext.PriceFloorModelGroup, account config.Account) ([]openrtb_ext.PriceFloorModelGroup, []error) {
 	var errs []error
 	var validModelGroups []openrtb_ext.PriceFloorModelGroup
+	if len(modelGroups) == 0 {
+		return validModelGroups, []error{fmt.Errorf("No model group present in floors.data")}
+	}
+
 	for _, modelGroup := range modelGroups {
+
+		if len(modelGroup.Schema.Fields) > account.PriceFloors.MaxSchemaDims {
+			errs = append(errs, fmt.Errorf("Invalid Floor Model = '%v' due to number of schema fields = '%v' are greater than limit %v", modelGroup.ModelVersion, len(modelGroup.Schema.Fields), account.PriceFloors.MaxSchemaDims))
+			continue
+		}
+
+		if len(modelGroup.Values) > account.PriceFloors.MaxRule {
```

Comment:
What if `MaxSchemaDims` or `MaxRule` is 0? I believe 0 means unbounded so it shouldn't result in an error.

Reply:
added `account.PriceFloors.MaxRule > 0` check before checking number of rules in request